src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -452,33 +513,47 @@ namespace EnergyPlus {
-                                                   this->sourceSideLocation.branchNum,
-                                                   this->sourceSideLocation.compNum);
+                if (this->waterSource) {
+                    rho = FluidProperties::GetDensityGlycol(
+                            DataPlant::PlantLoop(this->sourceSideLocation.loopNum).FluidName,
+                            DataGlobals::InitConvTemp,
+                            DataPlant::PlantLoop(this->sourceSideLocation.loopNum).FluidIndex,
+                            routineName);
+                    this->sourceSideDesignMassFlowRate = rho * this->sourceSideDesignVolFlowRate;
+                    PlantUtilities::InitComponentNodes(0.0,
+                                                       this->sourceSideDesignMassFlowRate,
+                                                       this->sourceSideNodes.inlet,
+                                                       this->sourceSideNodes.outlet,
+                                                       this->sourceSideLocation.loopNum,
+                                                       this->sourceSideLocation.loopSideNum,
+                                                       this->sourceSideLocation.branchNum,
+                                                       this->sourceSideLocation.compNum);
+                } else if (this->airSource) {
+                    rho = Psychrometrics::PsyRhoAirFnPbTdbW(DataEnvironment::StdBaroPress,
+                            DataEnvironment::OutDryBulbTemp,
+                            0.0,
+                            routineName);
+                    this->sourceSideDesignMassFlowRate = rho * this->sourceSideDesignVolFlowRate;
+                }
+
                 this->envrnInit = false;
             }
             if (!DataGlobals::BeginEnvrnFlag) {
                 this->envrnInit = true;
             }
         }
 
-        void EIRWaterToWaterHeatPump::getDesignCapacities(const PlantLocation &calledFromLocation,
-                                                          Real64 &MaxLoad,
-                                                          Real64 &MinLoad,
-                                                          Real64 &OptLoad) {
+        void EIRPlantLoopHeatPump::getDesignCapacities(const PlantLocation &calledFromLocation,
+                                                       Real64 &MaxLoad,
+                                                       Real64 &MinLoad,
+                                                       Real64 &OptLoad) {
             if (calledFromLocation.loopNum == this->loadSideLocation.loopNum) {
-                this->size();
+                this->sizeLoadSide();
+                if (this->waterSource) {
+                    this->sizeSrcSideWSHP();
+                } else if (this->airSource) {
+                    this->sizeSrcSideASHP();
+                }
```

Comment:
I like this, and I could see how in the long run _maybe_ someone could push this further to get:

```
if waterLoad:
  this->sizeLoadWater()
else:
  this->sizeLoadAir()
```